### PR TITLE
run test only if labels don't contain 'documentation'

### LINKED ISSUE #2226: Skip unit tests for docs-only changes
Shall we try to reduce the CI time for documentation-only changes?  I think of running just the docs CI when having the label `documentation` attached to the issue, and ignoring the whole pytest suite.

That would mean that we need to split our current pytest approach, where we test "normal" tests and documentation tests simultaneously.

This would greatly reduce the time to passing tests in minor (or large) documentation changes, and we would not be held up by increased memory usage, suddenly incompatible dependencies etc. like we were in the last few days.

(I'm thinking in particular of PRs like #2220)

### PR DESCRIPTION
**Pull request by nbiederbeck**
Fix #2226

### CODE REVIEW

**Comment by nbiederbeck** [comment]
Maybe `documentation-only` or `no-pytest-needed` would be a better label.

**Comment by kosack** [comment]
I rebased onto the latest main.  If this builds, we can merge

**Review by kosack — APPROVED** [review]

**Review by Tobychev — DISMISSED** [review]

**Comment by Tobychev** [comment]
I gave it another go, tried a different syntax for the conditional to see if that caused the tests to be marked as skipped instead of succeeded as [promised](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution) by the GitHub docs. I  even removed the "documentation" label, which I _really_ tough should ensure the tests are not skipped, but still get get marked as skipped and thus this will be stuck forever.

**Comment by Tobychev** [comment]
@kosack @nbiederbeck @maxnoe I finally figured out what the problem was: there's a [bug/design error](https://github.com/actions/runner/issues/952) in how required steps interact with matrix jobs and the conditional check: 
basically if you put the test at the top of a job like the initial PR did, you skip the parent job so the matrix of required jobs aren't even created, and thus they are not marked as skipped, and the requirement checker (or whatever its name is) will wait forever.

I've now implemented one of the two cludges proposed to get around this: put if gates into mostly every substep of the "test" workflow.

**Comment by LukasNickel** [comment]
> Maybe `documentation-only` or `no-pytest-needed` would be a better label.

:+1: on `documentation-only`.
I think it is not super clear, that a `documentation` tag would skip the code checks as a PR might contain changes in documentation and code.

The changes itself look fine to me. It is a bit confusing, that the skipped tests still show up in the list of checks, but its anyway helpful.

**Comment by Tobychev** [comment]
Changed the magic label to `documentation-only` and rebased on main to solve the problems with the docs failing to merge.

**Comment by Tobychev** [comment]
@maxnoe This could finally be merged once the doctests pass, but I don't really understand why they fail. Help!

**Review by kosack — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -75,28 +75,28 @@ jobs:
...
           PYTHON_VERSION: ${{ matrix.python-version }}
         run: |
           # setup correct python version
           sed -i -e "s/- python=.*/- python=$PYTHON_VERSION/g" environment.yml
 
       - name: mamba setup
-        if: matrix.install-method == 'mamba'
-        uses: mamba-org/setup-micromamba@v1
+        if: matrix.install-method == 'mamba' && contains(github.event.pull_request.labels.*.name, 'documentation-only') == false
+        uses: mamba-org/provision-with-micromamba@v14
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply .github/workflows/ci.yml:87]: It should be as before `mamba-org/setup-micromamba@v1` not `mamba-org/provision-with-micromamba@v14`

[the remark this answers is not in the document]
Comment (maxnoe) [reply .github/workflows/ci.yml:87]: I.e. only change the `if` line, not the `uses` line

[the remark this answers is not in the document]
Comment (Tobychev) [reply .github/workflows/ci.yml]: fixed.

**Review by maxnoe — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**